src/Shipment.php

```diff
@@ -36,47 +33,48 @@ public function __construct(OAuth2 $oAuth2, string $accessToken, String $refresh
 
         $this->refreshToken = $refreshToken;
 
-        $this->stack = HandlerStack::create();
-
-        $this->stack->push($this->middlewareRefreshToken());
+        $this->setHttp($this->makeClient());
+    }
 
-        $this->stack->push(
-            Middleware::retry($this->retryDecider(), $this->retryDelay())
-        );
+    public function setEnvironment(string $environment): void
+    {
+        parent::setEnvironment($environment);
 
-        $this->setHttp($this->client());
+        $this->setHttp($this->makeClient());
     }
 
-    public function client(): Client
+    public function makeClient(array $extraOptions = []): Client
     {
-        return new Client([
-            'handler' => $this->stack,
-            'base_uri' => Endpoint::ENDPOINTS[$this->getEnvironment()] . '/api/' . Endpoint::VERSIONS[$this->getEnvironment()] . '/',
+        $stack = $this->createStack();
```

Comment:
O que acha de deixar claro que esse stack se trata de um stack de middleware no nome da variável ?

```suggestion
        $middlewareStack = $this->createStack();
```

Reply:
Feito!